base/reduce.jl

```diff
@@ -861,3 +861,19 @@ function count(pred, a::AbstractArray)
     return n
 end
 count(itr) = count(identity, itr)
+
+function count(x::Array{Bool})
```

Comment:
```suggestion
function count(::typeof(identity), x::Array{Bool})
```

Let's override the more specific version here